compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/preserve-memo-validation/error.false-positive-extended-contextvar-scope.tsx

```diff
@@ -0,0 +1,32 @@
+// @validatePreserveExistingMemoizationGuarantees:true
+
+import {useCallback} from 'react';
+import {Stringify, useIdentity} from 'shared-runtime';
+
+/**
+ * Here, the *inferred* dependencies of cb are `a` and `t1 = LoadContext capture x_@1`.
+ * - t1 does not have a scope as it captures `x` after x's mutable range
+ * - `x` is a context variable, which means its mutable range extends to all
+ *    references / aliases.
+ * - `a`, `b`, and `x` get the same mutable range due to potential aliasing.
+ *
+ * We currently bail out because `a` has a scope and is not transitively memoized
+ * (as its scope is pruned due to a hook call)
```

Comment:
Interesting - looks like we are grouping the argument destructuring into a scope? that seems wrong, `a` and `b` aren't mutable so they shouldn't be getting added into any scopes.

Reply:
My understanding is that this is because `x` is a context variable, and since our understanding of mutable context variables is imprecise, `x` and all potential aliases get marked as mutable. I might be incorrect though -- can look into this more